### PR TITLE
[AC-1868] Fix UserCipherDetails_V2 Function

### PR DESCRIPTION
**Pull request by shane-melton**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

We introduced `UserCipherDetails_V2` in https://github.com/bitwarden/server/pull/3372 and we mistakenly inverted the logic surrounding the `[Edit]` and `[ViewPassword]` fields. This PR fixes that by re-introducing the `CASE WHEN .. THEN .. ELSE` statement for both of those fields.

## Code changes

Update the Db funciton and add a migration script.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Sql/Vault/dbo/Functions/UserCipherDetails_V2.sql [hunk 1]
```diff
@@ -13,8 +13,16 @@ WITH [CTE] AS (
 )
 SELECT
     C.*,
-    COALESCE(CU.[ReadOnly], CG.[ReadOnly], 0) AS [Edit],
-    COALESCE(CU.[HidePasswords], CG.[HidePasswords], 0) AS [ViewPassword],
+    CASE
+	    WHEN COALESCE(CU.[ReadOnly], CG.[ReadOnly], 0) = 0
+	    THEN 1
+	    ELSE 0
```
Comment (withinfocus) [inline src/Sql/Vault/dbo/Functions/UserCipherDetails_V2.sql:19]: ⛏️ Formatting.

**Review by withinfocus — COMMENTED** [review]
Waiting for team review first, but the DB change operationally seems fine.

File: src/Sql/Vault/dbo/Functions/UserCipherDetails_V2.sql [hunk 2]
```diff
@@ -13,8 +13,16 @@ WITH [CTE] AS (
...
+    CASE
+        WHEN COALESCE(CU.[ReadOnly], CG.[ReadOnly], 0) = 0
+        THEN 1
+        ELSE 0
+    END [Edit],
+    CASE
+    	WHEN COALESCE(CU.[HidePasswords], CG.[HidePasswords], 0) = 0
+    	THEN 1
+    	ELSE 0
+	END [ViewPassword],
```
Comment (withinfocus) [inline src/Sql/Vault/dbo/Functions/UserCipherDetails_V2.sql:25]: ⛏️ Drop one tab here 😜

Comment (shane-melton) [reply src/Sql/Vault/dbo/Functions/UserCipherDetails_V2.sql]: haha I guess Rider does not like to play well with SQL and tabs 🤦 

Fixed.

**Review by withinfocus — APPROVED** [review]
Still needs the team for logic review.

**Review by eliykat — APPROVED** [review]
Thanks for fixing this, I believe this was the only place where I got too clever and thought I could simplify the syntax. The other sprocs should be fine.

(You'll still need a vault team review though :( )

**Pull request merged**